network/endpoint_windows.go

```diff
@@ -79,11 +80,20 @@ func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
 		return nil, err
 	}
 
+	defer func() {
+		if err != nil {
+			log.Printf("[net] HNSEndpointRequest DELETE id:%v", hnsResponse.Id)
+			hnsResponse, err := hcsshim.HNSEndpointRequest("DELETE", hnsResponse.Id, "")
```

Comment:
Hopefully the IPAddress is also released from the IPAM following this somewhere in CNI

Reply:
yes it will free up..will test once.